util/threadpool_imp.cc

```diff
@@ -254,11 +254,25 @@ void* ThreadPoolImpl::Impl::BGThreadWrapper(void* arg) {
   size_t thread_id = meta->thread_id_;
   ThreadPoolImpl::Impl* tp = meta->thread_pool_;
 #ifdef ROCKSDB_USING_THREAD_STATUS
-  // for thread-status
-  ThreadStatusUtil::RegisterThread(
-      tp->GetHostEnv(), (tp->GetThreadPriority() == Env::Priority::HIGH
-                             ? ThreadStatus::HIGH_PRIORITY
-                             : ThreadStatus::LOW_PRIORITY));
+  // initialize it because compiler isn't good enough to see we don't use it
+  // uninitialized
+  ThreadStatus::ThreadType thread_type = ThreadStatus::NUM_THREAD_TYPES;
+  switch (tp->GetThreadPriority()) {
+    case Env::Priority::HIGH:
+      thread_type = ThreadStatus::HIGH_PRIORITY;
+      break;
+    case Env::Priority::LOW:
+      thread_type = ThreadStatus::LOW_PRIORITY;
+      break;
+    case Env::Priority::BOTTOM:
+      thread_type = ThreadStatus::BOTTOM_PRIORITY;
+      break;
+    case Env::Priority::TOTAL:
```

Comment:
Can this just be default: case? Then you can get rid of assert on line 274, and compiler will probably be okay with thread_type not being initialized..

Reply:
Yeah it's kind of a tradeoff. I wanted to avoid default case so that compiler error will happen if someone introduces a new `Env::Priority` and forgets to update this switch statement.

Reply:
Just curious, will -Wswitch-enum help? It is supposed to catch an unhandled enum value even with default present, but the option may be present only in newer versions.

Reply:
That's interesting, thanks! I tried it but found there are some places we are omitting a lot of enumerators, e.g., https://github.com/facebook/rocksdb/blob/master/db/dbformat.cc#L40.